app/jobs/socure_image_retrieval_job.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class SocureImageRetrievalJob < ApplicationJob
+  queue_as :default
+
+  attr_reader :document_capture_session_uuid
+
+  def perform(
+    document_capture_session_uuid:,
+    reference_id:,
+    image_storage_data:
+  )
+    @document_capture_session_uuid = document_capture_session_uuid
+
+    result = fetch_images(reference_id)
+    if result.is_a?(Idv::IdvImages)
+      result.write_with_data(image_storage_data:)
+    else
+      # removing the encryption keys from the failure data
+      failure_data = image_storage_data
+        .map { |k, v| v.select { |val| val.match(/file_id$/) } }
+        .reduce({}) { |a, b| a.merge(b) }
+      attempts_api_tracker.idv_image_retrieval_failed(
+        **failure_data,
+      )
```

Comment:
```suggestion
      attempts_api_tracker.idv_image_retrieval_failed(
        document_back_image_file_id: image_storage_data.dig(:back, :document_back_image_file_id),
        document_front_image_file_id: image_storage_data.dig(:front_front_image_file_id),
        document_passport_image_file_id: image_storage_data.dig(:passport, :document_passport),
        document_selfie_image_file_id: image_storage_data.dig(:selfie, :document_selfie),
      )
```

Slight preference for me towards being explicit about which keys we're trying to send here

Reply:
we've been able to avoid sending keys with `nil` value so i was adhering to that -- but am not super attached to it, especially in this scenario, so i can update

Reply:
i have sort of thought about that generally, and i think i am not quite right -- for instance in the idv events with more data in them, i think we have `nil` values for data not in the user pii bundle. in those cases it does seem like it would be potetntially useful to tell at a glance which are `nil`, as opposed to having to consult the specs. wdyt?